### PR TITLE
Fix duplicate keys in parallel prehandling

### LINKED ISSUE #8081: [Dynamic Region] TiFlash can't catch up with TiKV when there is only one big region pending prehandle
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### LINKED ISSUE #8180: [Dynamic Regions] tiflash oom and crash repeatedly after inject one of tiflash network partition last for 10mins
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run workload
go-tpc ch run -D tpcc --host tc-tidb.endless-ha-test-htap-tps-3030196-1-619 -P4000 --warehouses 2000 -T 32 --acThreads 1 --queries q13 --ignore-error '2013,1213,1105,1205,8022,8028,9004,9007,1062' --time 36000m --user root --password '' --interval '10s'
2、 inject one of tiflash network partition last for 10mins

### 2. What did you expect to see? (Required)
no oom and crash

### 3. What did you see instead (Required)
tiflash oom and crash repeatedly after fault recover
5ea4e562-7696-4309-91a4-596e41215f61

### 4. What is your TiFlash version? (Required)
git hash：952c9119b5513c2ea7ca47f7e2c6c932e164aaa8

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #8180 close #8081

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Debug/MockSSTReader.h [hunk 1]
```diff
@@ -80,13 +80,13 @@ struct MockSSTReader
...
         {
-            remained = iter != end;
             iter = begin;
+            remained = iter != end;
         }
         else if (et == EngineIteratorSeekType::Last)
         {
-            remained = iter != end;
             iter = end;
+            remained = iter != end;
```
Comment (JaySon-Huang) [inline dbms/src/Debug/MockSSTReader.h:89]: I think this means always `remained = false`?

Comment (CalvinNeo) [reply dbms/src/Debug/MockSSTReader.h:89]: fixed

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**